models/account.model.ts

```diff
@@ -49,6 +66,6 @@ UserSchema.pre<IUserModel>('save', async function (next) {
 
 });
 
-const User: Model<IUserModel> = mongoose.model<IUserModel>('Users', UserSchema);
+const Account: Model<IAccount> = mongoose.model<IAccount>('Account', AccountSchema);

```

Comment:
you are following very complex approach to create models based on mongoose. Please follow the simple example like this

 import { Schema, model } from 'mongoose';


const userSchema = new Schema({
  name: {
    type: String,
    required: true,
  },
  email: {
    type: String,
    required: true,
    unique: true,
  },
  password: {
    type: String,
    required: true,
  },
});

const User = model('User', userSchema);`

Reply:
model updated